include/cgimap/json_writer.hpp

```diff
@@ -49,8 +48,14 @@ class json_writer : public output_writer {
     yajl_gen_integer(gen, i);
   }
 
-  void entry(const char* s);
-  void entry(std::string_view sv);
+  template <typename T,
+            std::enable_if_t<std::is_convertible_v<T&&, std::string_view>,
+                             bool> = true>
+  void entry(T&& s)
+  {
+    auto sv = std::string_view(s);
```

Comment:
The idea here was to avoid char* ending up in entry(bool): https://godbolt.org/z/1PKz9o6dT

At least the generated code looks exactly as it should now. I need to check the std::forward<T> bit still.

Reply:
> The idea here was to avoid char* string literals ending up in entry(bool)

Ah, yeah then that's probably required to do it this way.

The generated code will _likely_ be the same even without std::forward since I don't believe std::string_view uses any optimization for rvalue moves.

Reply:
Yes, the optimizer even completely removes the intermediate string_view, and turns printing of string literals into something like:

```
   899f0:       48 8b 7d 08             mov    0x8(%rbp),%rdi
   899f4:       ba 1e 00 00 00          mov    $0x1e,%edx
   899f9:       48 8d 35 08 1b 07 00    lea    0x71b08(%rip),%rsi 
   89a00:       e8 5b 73 f9 ff          call   20d60 <yajl_gen_string@plt>
```